src/openms/include/OpenMS/KERNEL/MSExperiment.h

```diff
@@ -1286,8 +1331,81 @@ std::vector<MSChromatogram> extractXICs(
     std::vector<MSChromatogram > chromatograms_;
     /// spectra
     std::vector<SpectrumType> spectra_;
+    /// Spectrum range manager for tracking m/z, intensity, RT, and ion mobility ranges of spectra with MS level separation
+    SpectrumRangeManagerType spectrum_ranges_;
+    
+    /// Chromatogram range manager for tracking RT, intensity, and m/z ranges of chromatograms
+    ChromatogramRangeManagerType chromatogram_ranges_;
+    
+    /// Combined range manager that provides overall ranges across both spectra and chromatograms (maintained for backward compatibility)
+    RangeManagerType combined_ranges_;
 
-private:
+  public:
+    /**
+     * @brief Returns a reference to the spectrum range manager
+     *
+     * The spectrum range manager provides access to m/z, intensity, retention time, and ion mobility
+     * ranges for spectra, with separate tracking for different MS levels.
+     *
+     * @return Reference to the spectrum range manager
+     * @see SpectrumRangeManager
+     */
+    SpectrumRangeManagerType& spectrumRanges() { return spectrum_ranges_; }
```

Comment:
this would allow 'hacking' ranges externally and destroy the invariant.
Do we really want/need this?

Reply:
ah that is good point. ranges are already pretty lax regarding invariants. Will remove